CONTRIBUTING.md

```diff
@@ -120,11 +132,37 @@ the respective change set.
 The length of time required for peer review is unpredictable and will vary from
 pull request to pull request.
 
+Rebasing Pull Requests
+-------------------------
+It may become necessary for a pull request to be rebased after other pull requests have been
+merged. This is typically due to mutually exclusive changes (conflicts) between your pull
+request and the current `master` branch.
+
+When a rebase is needed, a comment will be added to the pull request indicating this need.
+Rather than simply merge the `master` branch into your pull request (which results in an
+ugly and confusing merge commit), it is better to use git's rebase feature. The basic
+workflow is as follows:
+
+    # replace 'origin' with the remote name for the main project repo in the example
+    git checkout your_branch_name
+    git fetch origin
+    git pull --rebase origin master
+
+This will "rewind" your branch commits, pull any new commits from `master`, then attempt to
+re-apply your commits on top of the new HEAD. If any conflicts are found, the process will
+pause and allow you to resolve any conflicts. Once conflicts have been resolved:
+
+    git rebase --continue
+
+Repeat as necessary until there are no more conflicts and your git tree is in a clean state.
+The final step is to push you rebased branch back up to github:
+
+    git push -f # force pushes the branch to github
 
 Pull Request Philosophy
 -----------------------
 
-Patch sets should always be focused. For example, a pull request could add a
+Patchsets should always be focused. For example, a pull request could add a
```

Comment:
[consistency] This changed "Patch sets" to "Patchsets"; yet 3 other occurrences of "patch set" exist